app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,58 @@
+class QuestionsController < ApplicationController
+  include Votable
+
+  def index
+    @questions = Question.includes(:answers, :comments).page(page).per(limit)
+
+    render json: @questions
+  end
+
+  private
+
+  def manager
+    @manager ||= QuestionManager.new(resource, user)
+  end
+
+  def klass
+    @klass ||= Question
+  end
+
+  def valid_index
+    param! :page, Integer, required: false, default: 0
+    param! :limit, Integer, required: false, default: 10
+  end
+
+  def valid_create
+    param! :heading, String, required: true, message: 'Question heading not specified'
+    param! :description, String, required: true, message: 'Question description not specified'
+    param! :user_id, Integer, required: true
+  end
+
+  def valid_show
+    param! :id, Integer, required: true
+  end
+
+  def valid_update
+    param! :id, Integer, required: true
+    param! :heading, String, required: true, message: 'Question heading not specified'
+    param! :description, String, required: true, message: 'Question description not specified'
+    param! :user_id, Integer, required: true
+  end
+
+  def valid_destroy
+    param! :id, Integer, required: true
+    param! :user_id, Integer, required: true
+  end
+
+  def params_attributes
+    params.permit(:heading, :description)
+  end
+
+  def page
+    @page ||= params[:page]
```

Comment:
default values?

Reply:
provided in param!

https://github.com/lokendrasingh-1mg/StackOverflowRailsClone/blob/a8e893c805f36fde13c98816244288767f3a7327/app/controllers/questions_controller.rb#L20-L22